src/System.Net.Http/src/System/Net/Http/HttpResponseMessage.cs

```diff
@@ -115,6 +116,20 @@ public HttpResponseHeaders Headers
             }
         }
 
+        public HttpResponseHeaders TrailingHeaders
+        {
+            get
+            {
+                // If trailing headers are not ready yet, return null.
+                return _trailingHeaders;
+            }
+            set
```

Comment:
It seems weird that you can set this property. We don't allow you to set the Headers property.

Reply:
There shouldn't be a setter for this property.

Reply:
I posted the reason [below](https://github.com/dotnet/corefx/pull/35337#discussion_r257106595). I used the setter to reduce allocation. Let me think if there is an alternative way to achieve it.

Reply:
If you have to have a setter, then it should not be a public property.  It should be internal only.